test_my_class.cpp

```diff
@@ -1,47 +1,880 @@
-//@file: test_my_class.cpp
-//@author: Veselina Velkova (veselina.velkova@strypes.eu)
-//@brief: This file contains all unit test cases.
-//@version: 0.1
-//@date: 21.04.2023
-
+#include<limits>
+#include<cmath>
+#include<cstdlib>
+#include<algorithm>
 #include<iostream>
 #include "my_class.h"
+#include "test_my_class.hpp"
+
+#define _USE_MATH_DEFINES
 
 #define IS_TRUE(x) {if(!(x)) std::cout<<__FUNCTION__<<" failed on line "<<__LINE__<<std::endl;}
 
-#define TEST_CLASS_POSITIVE_DOUBLE(3.141592653589793238462643383279502884197169)
-#define TEST_CLASS_NEGATIVE_DOUBLE(-3.141592653589793238462643383279502884197169)
+#define TEST_CLASS_POSITIVE_DOUBLE (M_PI)
+#define TEST_CLASS_NEGATIVE_DOUBLE (-M_PI)
+
+#define TEST_CLASS_POSITIVE_FLOAT (5.5)
+#define TEST_CLASS_NEGATIVE_FLOAT (-3.5)
+#define TEST_CLASS_BIG_POSITIVE_FLOAT (14.3)
+
+#define TEST_CLASS_POSITIVE_INT (6)
+#define TEST_CLASS_NEGATIVE_INT (-4)
+
+#define TEST_CLASS_NOT_THAT_LARGE_POS_INT (31650)
+#define TEST_CLASS_NOT_THAT_LARGE_NEGATIVE_INT (-31650)
+
+#define TEST_CLASS_LARGE_POSITIVE_INT (std::numeric_limits<int>::max())
+#define TEST_CLASS_LARGE_NEGATIVE_INT (std::numeric_limits<int>::min())
 
-#define TEST_CLASS_POSITIVE_FLOAT(5.5)
-#define TEST_CLASS_NEGATIVE_FLOAT(-3.5)
+#define TEST_CLASS_UNSIGNED_LONG_LONG_INT (std::numeric_limits<unsigned long long int>::max())
```

Comment:
stupid qn: what's wrong with INT_MAX?

Reply:
Oh I think it should work as well, but unsigned long long int has max value 18, 446, 744, 073, 709, 551, 615 and INT_MAX (64bit integer) is 9,223,372,036,854,775,807.

Reply:
ok, fair enough. I have been using INT_MAX for int types, but the numerical limits also work.